cmd/install.go

```diff
@@ -0,0 +1,43 @@
+package cmd
+
+import (
+	"context"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin/resolver"
+)
+
+var installUpgrade bool
+
+var installCmd = &cobra.Command{
+	Use:   "install",
+	Short: "Install plugins",
+	Long:  "Install Fabric plugins",
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		var diags diagnostics.Diag
+		eval := NewEvaluator()
+		defer func() {
+			err = eval.Cleanup(diags)
+		}()
+		diags = eval.ParseFabricFiles(os.DirFS(cliArgs.sourceDir))
+		if diags.HasErrors() {
+			return
+		}
+		if diags.Extend(eval.LoadPluginResolver(true)) {
+			return
+		}
+		lockFile, stdDiags := eval.Resolver.Install(context.Background(), eval.LockFile, installUpgrade)
```

Comment:
You can use `cmd.Context()` instead of `context.Background()` here. It's a `ctrl+c` cancelable context.

It also has a cleanup-context that is canceled only by a second `ctrl+c`, similar to docker compose two-stage graceful/hard shutdown, useful for defered cleanup tasks that also take context. [/pkg/clicontext/clicontext.go](https://github.com/blackstork-io/fabric/blob/0df838596d4ebf3843d08bede2f6a6e7fdf087da/pkg/clicontext/context.go) has the code

Reply:
Ahh, I see. I will use `cmd.Context()` then.
I did not notice that you use ExecuteContext from cobra at first since`main.go` looked like this:
```go
func main() {
	cmd.Execute()
}
```
I think we should move graceful shutdown logic to main.go and pass context from there. It's more readable that way than hiding it under sub-package. Maybe on some other PR when you are doing some refactoring.